src/query-builder/aggregate-function-builder.ts

```diff
@@ -0,0 +1,114 @@
+import { freeze } from '../util/object-utils.js'
+import { AggregateFunctionNode } from '../operation-node/aggregate-function-node.js'
+import { OperationNodeSource } from '../operation-node/operation-node-source.js'
+import { AliasNode } from '../operation-node/alias-node.js'
+import { IdentifierNode } from '../operation-node/identifier-node.js'
+import { preventAwait } from '../util/prevent-await.js'
+import { OverBuilder } from './over-builder.js'
+import { createOverBuilder } from '../parser/parse-utils.js'
+
+export class AggregateFunctionBuilder<DB, TB extends keyof DB, O = unknown>
+  implements OperationNodeSource
+{
+  readonly #props: AggregateFunctionBuilderProps
+
+  constructor(props: AggregateFunctionBuilderProps) {
+    this.#props = freeze(props)
+  }
+
+  /**
+   * TODO: ...
+   */
+  as<A extends string>(
+    alias: A
+  ): AliasedAggregateFunctionBuilder<DB, TB, O, A> {
+    return new AliasedAggregateFunctionBuilder(this, alias)
+  }
+
+  /**
+   * TODO: ...
+   */
+  distinct(): AggregateFunctionBuilder<DB, TB, O> {
+    return new AggregateFunctionBuilder({
+      ...this.#props,
+      aggregateFunctionNode: AggregateFunctionNode.cloneWithDistinct(
+        this.#props.aggregateFunctionNode
+      ),
+    })
+  }
+
+  /**
+   * TODO: ...
+   */
+  over(
+    over?: OverBuilderCallback<DB, TB>
+  ): AggregateFunctionBuilder<DB, TB, O> {
+    const builder = createOverBuilder()
+
+    return new AggregateFunctionBuilder({
+      ...this.#props,
+      aggregateFunctionNode: AggregateFunctionNode.cloneWithOver(
+        this.#props.aggregateFunctionNode,
+        (over ? over(builder) : builder).toOperationNode()
+      ),
+    })
+  }
+
+  toOperationNode(): AggregateFunctionNode {
+    return this.#props.aggregateFunctionNode
+  }
+}
+
+/**
+ * {@link AggregateFunctionBuilder} with an alias. The result of calling {@link AggregateFunctionBuilder.as}.
+ */
+export class AliasedAggregateFunctionBuilder<
+  DB,
+  TB extends keyof DB,
+  O = unknown,
+  A extends string = never
+> implements OperationNodeSource
+{
+  readonly #aggregateFunctionBuilder: AggregateFunctionBuilder<DB, TB, O>
+  readonly #alias: A
+
+  /**
+   * @private
+   *
+   * This needs to be here just so that the typings work. Without this
+   * the generated .d.ts file contains no reference to the type param A
+   * which causes this type to be equal to AliasedRawBuilder with any A
+   * as long as O is the same.
+   */
+  protected get alias(): A {
+    return this.#alias
+  }
+
+  toOperationNode(): AliasNode {
+    return AliasNode.create(
+      this.#aggregateFunctionBuilder.toOperationNode(),
+      IdentifierNode.create(this.#alias)
+    )
+  }
+
+  constructor(
```

Comment:
Move the constructor before getters and methods.

Reply:
Good catch! done.